### DISCUSSION TITLE
Add support for Square's payment processing API

### DISCUSSION THREAD

**Discussion by codyjames** [craftcms/commerce#2485] [Plugin ideas]
### Description
Square has a payment processing API. Using this instead of Stripe (for instance) would be nice for clients who use Square already. Among other things, it would allow them to keep all of their reporting in one system.

More information: https://squareup.com/us/en/developers/online-payment-apis

**Comment by codyjames** [comment]
My client is looking to sell CBD and Square has been one of the few major payment processors willing to work with CBD companies.

From my client:

> "Yeah I noticed there wasn’t a native option or 3rd party plugin. Square is one of only a few major payment processors willing to work with CBD companies. The others that I know of are emerchantbroker, digipay, and potentially authorize.net but that’s not confirmed. There are some other small processors available but we’re trying to stick with a more established/recognizable company.
> 
> Craft should really consider targeting CBD ecomm companies as there is a serious gap in that market.

**Comment by ax2000** [comment]
Hi guys, just wondering if this is part of your plans. I've also got a client keen to use square as payment gateway.

**Comment by joshuabaker** [comment]
@ax2000 We’re working on a [Square plugin](https://github.com/craftplugins/commerce-square) currently. Should be in public beta within the next 4–6 weeks.

**Comment by joshuabaker** [comment]
@codyjames @ax2000 The [Square plugin public beta](https://github.com/craftplugins/square) is now available to install via composer (`composer require craftplugins/square`).

**Comment by PhilRedm** [comment]
Any update on when this might hit the plugin store? I've got a client that's currently on Square that could be moved to Craft Commerce when it's available.

**Comment by joshuabaker** [comment]
Hey @PhilRedm. Not far off ready for the Plugin Store. I had to rename the package at P&T’s request. There’s nothing stopping you using [the package](https://packagist.org/packages/augmentations/craft-commerce-square) in the meantime though.
